sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -4082,18 +4082,21 @@ object ApplyCharTypePadding extends Rule[LogicalPlan] {
 
         // String literal is treated as char type when it's compared to a char type column.
         // We should pad the shorter one to the longer length.
-        case b @ BinaryComparison(e @ AttrOrOuterRef(attr), lit) if lit.foldable =>
+        case b @ BinaryComparison(e @ AttrOrOuterRef(attr), lit)
```

Comment:
do you know why it's a problem only for partition predicates, not normal predicates?

Reply:
sorry, maybe my description is not very precise, the problem exists for char type column predicate, it is irrelevant to whether the partition is used.

Reply:
Can you clarify it in your PR description and update tests accordingly?

Reply:
ok, i'll update, thank you.